Dockerfile

```
@@ -19,28 +23,96 @@ COPY ./app/img/ /source/app/img/
 
-# Copy lunaria build output
-COPY --from=builder /source/app /export
+# --------------------------------------------------------
+# Publish Script: Option to host app locally or on nft.storage
+# --------------------------------------------------------
+
+COPY <<'EOF' /entrypoint.sh
+#!/bin/sh
+set -e
+
+if [ $# -ne  1 ]; then
+  echo "Example usage: docker run --rm ghcr.io/darkflorist/lunaria:latest [docker-host|nft.storage]"
+  exit  1
+fi
+
+case $1 in
+
+  docker-host)
+    # Show the IFPS build hash
+    echo "Build Hash: $(cat /ipfs_hash.txt)"
 
-# add the build output to IPFS and write the hash to a file
-RUN ipfs add --cid-version 1 --quieter --only-hash --recursive /export > ipfs_hash.txt
+    # Determine the IPV4 address of the docker-hosted IPFS instance
+    IPFS_IP4_ADDRESS=$(getent ahostsv4 host.docker.internal | grep STREAM | head -n 1 | cut -d ' ' -f 1)
 
-# print the hash for good measure in case someone is looking at the build logs
-RUN cat ipfs_hash.txt
+    echo "Adding files to docker running IPFS at $IPFS_IP4_ADDRESS"
+		IPFS_HASH=$(ipfs add --api /ip4/$IPFS_IP4_ADDRESS/tcp/5001 --cid-version 1 --quieter -r /export)
+    echo "Uploaded Hash: $IPFS_HASH"
+    ;;
 
-# this entrypoint file will execute `ipfs add` of the build output to the docker host's IPFS API endpoint, so we can easily extract the IPFS build out of the docker image
-RUN printf '#!/bin/sh\nipfs --api /ip4/`getent ahostsv4 host.docker.internal | grep STREAM | head -n 1 | cut -d \  -f 1`/tcp/5001 add --cid-version 1 -r /export' >> entrypoint.sh
-RUN chmod u+x entrypoint.sh
+  nft.storage)
+    if [ -z $NFTSTORAGE_API_KEY ] || [ $NFTSTORAGE_API_KEY = "" ]; then
+      echo "NFTSTORAGE_API_KEY environment variable is not set";
+      exit  1;
+    fi
+
+    # Show the IFPS build hash
+    echo "Build Hash: $(cat /ipfs_hash.txt)"
+
+    # Create a CAR archive from build hash
+    echo "Uploading files to nft.storage..."
+		IPFS_HASH=$(ipfs add --cid-version 1 --quieter -r /export)
+    ipfs dag export $IPFS_HASH > output.car
+
+    # Upload the entire directory to nft.storage
+		UPLOAD_RESPONSE=$(curl \
+			-X POST -H "Authorization: Bearer $NFTSTORAGE_API_KEY" \
+			-H "Content-Type: application/car" \
+			--data-binary @output.car \
+			-s \
+			https://api.nft.storage/upload)
+
+		# Show link to nft.storage (https://xxx.ipfs.nftstorage.link)
+		UPLOAD_SUCCESS=$(echo "$UPLOAD_RESPONSE" | jq -r ".ok")
+
+		if [ "$UPLOAD_SUCCESS" = "true" ]; then
+			echo "Succesfully uploaded to https://"$(echo "$UPLOAD_RESPONSE" | jq -r ".value.cid")".ipfs.nftstorage.link"
+		else
+			echo "Upload Failed: " $(echo "$UPLOAD_RESPONSE" | jq -r ".error.message")
+		fi
+    ;;
+
+  *)
+    echo "Invalid option: $1"
+    echo "Example usage: docker run --rm ghcr.io/darkflorist/lunaria:latest [docker-host|nft.storage]"
+    exit  1
+    ;;
+esac
+EOF
+
+RUN chmod u+x /entrypoint.sh
 
 ENTRYPOINT [ "./entrypoint.sh" ]
```

These two lines need to reference the same file.  One refers to `/entrypoint.sh` and the other refers to `./entrypoint.sh`.  It appears the file is written to `/entrypoint.sh`, so this is probably what you want:
```suggestion
RUN chmod u+x /entrypoint.sh

ENTRYPOINT [ "/entrypoint.sh" ]
```

---

We should also specify the required environment variable of `NFTSTORAGE_API_KEY` with:
```
ENV NFTSTORAGE_API_KEY
```
This will make it so tools will prompt the user for the variable rather than the user needing to guess it is necessary.